src/regex.rs

```diff
@@ -139,3 +160,26 @@ fn is_trivial_regex(s: &regex_syntax::Expr) -> Option<&'static str> {
         _ => None,
     }
 }
+
+struct RegexVisitor<'v, 't: 'v> {
+    cx: &'v LateContext<'v, 't>,
+    spans: HashSet<Span>,
+}
+
+impl<'v, 't: 'v> Visitor<'v> for RegexVisitor<'v, 't> {
+    fn visit_expr(&mut self, expr: &'v Expr) {
+        if let Some(span) = is_expn_of(self.cx, expr.span, "regex") {
```

Comment:
Can we limit this to a single type of expression that will always be found in a regex macro expn? `with_expn_info` is an index of a very large uncached vector, so it could be slow.

(We do have much slower lints, but if this can be optimized, it should be)